### DISCUSSION TITLE
Heroku-like Sleep and Awaken upon Request for Idle Containers

### DISCUSSION THREAD

**Discussion by ScottGuthart** [dokku/dokku#4443] [Q&A]
Hey there,

I'd like my apps to behave similarly to free apps on Heroku, where if no requests are made for a period of time they are removed from memory/go to sleep until the next request.

How would I do this with Dokku?

**Comment by josegonzalez** [comment]
I think we'd need an out-of-band process manager to do this. You'd have to listen for requests to the domain - via [access_by_lua](https://github.com/openresty/lua-nginx-module#access_by_lua) if using openresty, or probably `auth_request` via nginx - and turn on the app, then allow the request to go through. You'd also probably need to change the upstreams to stream to something else that would dynamically proxy to the new hosts, as otherwise the request would fall on it's head.

The annoying part is `ps:start` is a bit slow atm because we still build the container - though that might be something we can avoid soon - but its certainly doable.

Lastly, the above would only apply to the docker-local scheduler. We'd probably need something different for Kubernetes or Nomad, though at least the former probably has _something_ in the community to handle this sort of behavior.

At the moment, I would say this isn't straightforward to do in Dokku, but worth exploring if its something you have the time for. I'd also be interested in investigating it via sponsorship if you/your org requires this.
